### DISCUSSION TITLE
How to get one store to react to changes in another?

### DISCUSSION THREAD

**Discussion by zachbryant** [vuejs/pinia#1669] [Help and Questions]
I have two stores, A and B. I want B to react to changes in A's state.

A has an action `fetchData` that completes as expected, and a getter/setter pair `activeObject`. User navigation dictates `activeObject`. B has separate concerns but derives some state from A in its own getter, `bFromA`.

The docs on [accessing another store's getter](https://pinia.vuejs.org/core-concepts/getters.html#accessing-other-stores-getters) say you can "directly _use it_", but in practice this isn't working. My components that consume `bFromA` don't react to changes in `activeObject` based on the code below. If the hmr updates without a complete page reload, I do see the expected state change.

it seems the only way to react to changes from A in B is to follow the [action subscription approach](https://pinia.vuejs.org/core-concepts/actions.html#subscribing-to-actions) and subscribe to `fetchData`. This seems completely counter to the docs, and what I think Pinia should be able to do without subscriptions. What's wrong with my approach?

In reality, I have more stores than this. Merging A and B is not an option.

Store A
```typescript
export const useStoreA = defineStore('storeA', {
  state: () => ({
    _activeObject: { someProp: true },
  }),
  getters: {
    activeObject: (state) => state._activeObject,
  },
  actions: {
    fetchData() { 
      /* network call */ 
    },
    setActiveObject(obj) {
      this._activeObject = obj;
    }
  }
});
```

Store B
```typescript
export const useStoreA = defineStore('storeB', {
  state: () => ({
    /* some values */
  }),
  getters: {
    bFromA(state) {
      const storeA = useStoreA();
      /* have tried both computed and plainly using storeA.activeObject */
      const { activeObject } = storeToRefs(storeA);
      return activeObject.value?.someProp ?? true;
    }
  },
});
```

Component consuming B
```typescript
setup() {
  const storeB = useStoreB();
  return {
    bFromA: computed(() => storeB.bFromA);
  }
}
```

[Also Asked on SO](https://stackoverflow.com/questions/73797801/pinia-respond-to-changes-in-one-store-from-another)

**Comment by posva** [comment]
This can be achieved in option stores by adding a `$subscribe()` within a custom pinia plugin but I think the most convenient way is to just use a watcher within a setup store
